src/com/boxboat/jenkins/library/buildVersions/GitBuildVersions.groovy

```diff
@@ -84,28 +116,50 @@ class GitBuildVersions implements Serializable {
         """
     }
 
-    def setRepoEventVersion(String gitRemotePath, String gitTagPrefix, String event, SemVer semVer) {
+    def setRepoEventVersion(String gitRemotePath, String gitTagPrefix, String event, SemVer semVer, Map<Object, Object> metadata = null) {
         def eventWithPrefix = gitTagPrefix ? "${gitTagPrefix}${event}" : event
         def dir = "${gitRepo.dir}/repo-versions/${gitRemotePath}"
-        def path = "${dir}/${Utils.alphaNumericDashLower(eventWithPrefix)}.txt"
-        Config.pipeline.sh """
-            mkdir -p "${dir}"
-            echo '${semVer.toString()}' > "${path}"
-        """
+        def path = "${dir}/${Utils.alphaNumericDashLower(eventWithPrefix)}"
+        def yamlFile = "${path}.yaml"
+        def txtFile = "${path}.txt"
+
+        // remove text file for conversion from text to yaml
+        if (Config.pipeline.fileExists(txtFile)) {
+            Config.pipeline.sh """
+                rm "${txtFile}"
+            """
+        }
+        Map<String, Object> data = new LinkedHashMap<>()
+        data.put("version", semVer.toString())
+        data.put("previousVersion", semVer.getPreviousVersion())
+        if(metadata != null && !metadata.isEmpty()) {
+            data.put("metadata", metadata)
+        }
+        def yamlStr = YamlUtils.dump(data)
```

Comment:
In `setEventImageVersion` you kept the `mkdir -p "${dir}"` but here it's removed - is that intentional? I'm not totally certain but I thought `writeFile` would create the directory structure if needed...

Reply:
Good catch... I should put that back